snuba/datasets/errors.py

```diff
@@ -201,18 +202,6 @@ def column_expr(
             column_name, query, parsing_context, table_alias
         )
 
-    def _get_promoted_columns(self) -> Mapping[str, FrozenSet[str]]:
-        return {
-            "tags": frozenset(self.__promoted_tag_columns.values()),
-            "contexts": frozenset(),
-        }
-
-    def _get_column_tag_map(self) -> Mapping[str, Mapping[str, str]]:
-        return {
-            "tags": {col: tag for tag, col in self.__promoted_tag_columns.items()},
-            "contexts": {},
-        }
```

Comment:
THis way we can remove this stuff from the dataset top level and move it where it belongs which is on the TableSource (since columns are defined per table source).